File: tests/factories/product.py
import factory

from salsa import models

from .base import ModelFactory
from .user_account import UserAccountFactory


class ProductFactory(ModelFactory):
    active = True
    name = factory.Faker('text')
    description = factory.Faker('text')
    image_urls = factory.sequence(lambda n: 'image_url_{0}'.format(n))
    user_account = factory.SubFactory(UserAccountFactory)

    class Meta:
        model = models.Product

File: flutter/helistrong/lib/models/product.dart
import 'package:helistrong/models/review.dart';
import 'package:helistrong/models/user.dart';

class Product {
  final String id;
  final bool active;
  final String name;
  final DateTime updatedAt;
  final String description;
  final DateTime createdAt;
  final double avgNumStars;
  final List<String> imageUrls;
  final User user;
  final List<Review> reviews;

  Product(this.id, this.active, this.name, this.updatedAt, this.description, this.createdAt, this.avgNumStars, this.imageUrls, this.user, this.reviews);
}
